src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -58,8 +76,11 @@
   private String description;
   private String frameworkName;
   private String slavesUser;
-  private String principal;
-  private String secret;
+  private String credentialsId;
+  @Deprecated
+  private transient String principal;
+  @Deprecated
+  private transient String secret;
```

Comment:
Looking at the `readResolve` method, it seems `principal` and `secret` get reset to `null` after being used to resolve a `credentialsId`. Is there a need to track calls to the setters methods for these properties and then run the equivalent of the `readResolve` again? I assume the fact that the setter methods exist may mean someone can call them.